packages/engine/lib/execution/src/runner/javascript/mod.rs

```diff
@@ -885,13 +885,16 @@ impl<'s> ThreadLocalRunner<'s> {
         &self,
         data_ptr: NonNull<T>,
         data_len: usize,
-        _data_capacity: usize, // for future use to create a `Buffer::from_raw_parts`
         target_len: usize,
     ) -> Buffer {
-        // TODO: OPTIM: We currently copy the buffers because the JavaScript representation of
-        //   arrays does not match the Rust implementation. Try to reduce copies where possible by
-        //   reusing it, i.e. check, if `target_len` >= `data_capacity` and constructing it from raw
-        //   parts.
+        if data_len == target_len {
+            return Buffer::from_custom_allocation(
+                data_ptr.cast::<u8>(),
+                data_len * std::mem::size_of::<T>(),
+                Arc::new(()),
```

Comment:
This is supposed to be the allocator which owns the data of this `Buffer`. Since JS is taking care of freeing it and Rust should just do nothing I think this is the simplest thing to do.

Reply:
Sounds reasonable